pkg/ssh/ssh.go

```diff
@@ -27,15 +28,19 @@ func Dial(options *ConnectionDialOptions, kind EngineMode) (*ssh.Client, error)
 }
 
 func Exec(options *ConnectionExecOptions, kind EngineMode) (string, error) {
+     return ExecWithInput(options, kind, nil)
+}
+
+func ExecWithInput(options *ConnectionExecOptions, kind EngineMode, input *os.File) (string, error) {
```

Comment:
input should be io.Reader because the lower level APIs accept an io.Reader and this is much more flexible.

Reply:
That makes sense to me.  Thanks.

Reply:
(I can't tell if the CI tests are meaningful at this point.  I'll manually test podman-image-scp with these changes as soon as I get a chance.)

Reply:
I don't think there are any useful scp tests here.